web/src/reducers/header.ts

```diff
@@ -0,0 +1,45 @@
+const SETLOGIN = 'SETLOGIN' as const;
+const SETSUBPAGE = 'SETSUBPAGE' as const;
+
+export const setLogin = (payload: boolean) => ({
+  type: SETLOGIN,
+  payload: !payload,
+});
+
+export const setSubPageOpen = (payload: boolean) => ({
+  type: SETSUBPAGE,
+  payload: !payload,
+});
+
+export type HeaderAction = ReturnType<typeof setLogin> | ReturnType<typeof setSubPageOpen>;
+
+export const initialAction: HeaderAction = {
+  type: 'SETSUBPAGE',
+  payload: false,
+};
+
+export interface HeaderState {
+  isLogin: boolean;
+  isSubPageOpen: boolean;
+}
+
+export const initialState: HeaderState = {
+  isLogin: false,
+  isSubPageOpen: false,
+};
+
+/**
```

Comment:
Please add the purpose of this reducer.

And mark the type with `{}` as follows:

```typescript
/**
 * <purpose of this reducer>
 * 
 * @param {HeaderState} state State value of Check Login and Opened SubPage
 * @param {HeaderAction} action Action method for setLogin and setSubPageOpen
 * @returns {HeaderState} state Updated state value of Check Login and Opened SubPage
 */
```